exonum-java-binding/core/rust/src/storage/list_index.rs

```diff
@@ -73,12 +69,10 @@ pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ListIndexPro
     index: jlong,
 ) -> jbyteArray {
     let res = panic::catch_unwind(|| {
-        let val = match *handle::cast_handle::<IndexType>(list_handle) {
-            IndexType::SnapshotIndex(ref list) => list.get(index as u64),
-            IndexType::ForkIndex(ref list) => list.get(index as u64),
-        };
-        match val {
-            Some(val) => env.byte_array_from_slice(&val),
+        let list = handle::cast_handle::<Index>(list_handle);
+        let value = list.get(index as u64);
+        match value {
+            Some(value) => env.byte_array_from_slice(&value),
             None => Ok(ptr::null_mut()),
         }
```

Comment:
Such pattern appears everywhere, would it make sense extracting `optional_value_to_java(Option<T>): jbytearray` (`optional_bytes_to_java`, `optional_slice_to_java` — whatever makes most sense)?